crates/scarb-api/src/lib.rs

```diff
@@ -75,64 +78,121 @@ fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
     Ok(starknet_artifacts)
 }
 
+#[derive(PartialEq, Debug)]
+struct ContractArtifactData {
+    path: Utf8PathBuf,
+    test_type: Option<String>,
+}
+
+fn build_target_path(target_dir: &Utf8Path, file_name: &str) -> Option<Utf8PathBuf> {
+    let path = target_dir.join(file_name);
+
+    if path.exists() {
```

Comment:
I think the path should always exist in the places you use it, otherwise there is an error in our logic (we forget to call scarb/ignore its failure) or scarb doesn't work, in which case we should error anyways I think